src/components/grid/grid.test.js

```diff
@@ -0,0 +1,360 @@
+        </GridItem>
+      </GridContainer>
+    </>
+  );
+};
+
+context("Testing Grid component", () => {
+  describe("should render Grid component", () => {
+    it.each([
+      [1, "auto", "auto"],
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to default and verify pod width and height",
+      (index, colStart, colEnd) => {
+        CypressMountWithProviders(<GridComponent />);
+
+        ViewportSize("default");
+        if (index === 1) {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        } else {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        }
+      }
+    );
+
+    it.each([
+      [1, "auto", "auto"],
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to small and verify pod width and height",
+      (index, colStart, colEnd) => {
+        CypressMountWithProviders(<GridComponent />);
+
+        ViewportSize("extra small");
+        if (index === 1) {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        } else {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        }
+      }
+    );
+
+    it.each([
+      [1, "auto", "auto"],
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to medium and verify pod width and height",
+      (index, colStart, colEnd) => {
+        CypressMountWithProviders(<GridComponent />);
+
+        ViewportSize("medium");
+        if (index === 1) {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        } else {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        }
+      }
+    );
+
+    it.each([
+      [1, "auto", "auto"],
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to large and verify pod width and height",
+      (index, colStart, colEnd) => {
+        CypressMountWithProviders(<GridComponent />);
+
+        ViewportSize("large");
+        if (index === 1) {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
```

Comment:
Same query here about the possible duplication of assertion in the `if` and `else` as in line 78 onwards.